R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -1,147 +1,322 @@
+      verify_in_unit_interval("percentile", percentile, TRUE)
+      verify_string("by", by, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$percentile_by(percentile, by)))
+    },
+    
+    count_by = function(col = "n", by = character()) {
+      verify_string("col", col, TRUE)
+      verify_string("by", by, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$count_by(col, by)))
+    },
+    
+    cross_join = function(right_side, columns_to_match = character(), columns_to_add = character()) {
+      verify_string("columns_to_match", columns_to_match, FALSE)
+      verify_string("columns_to_add", columns_to_add, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$cross_join(
+        right_side$.internal_rcpp_object,
+        columns_to_match, columns_to_add
+      )))
+    },
+    
+    natural_join = function(right_side, columns_to_match = character(), columns_to_add = character()) {
+      verify_string("columns_to_match", columns_to_match, FALSE)
+      verify_string("columns_to_add", columns_to_add, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$natural_join(
+        right_side$.internal_rcpp_object,
+        columns_to_match, columns_to_add
+      )))
+    },
+    
+    exact_join = function(right_side, columns_to_match = character(), columns_to_add = character()) {
+      verify_string("columns_to_match", columns_to_match, FALSE)
+      verify_string("columns_to_add", columns_to_add, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$exact_join(
+        right_side$.internal_rcpp_object,
+        columns_to_match, columns_to_add
+      )))
+    },
+    
+    sort = function(by, descending = FALSE, abs_sort = FALSE) {
+      verify_string("by", by, FALSE)
+      verify_bool("descending", descending, FALSE)
+      verify_bool("abs_sort", abs_sort, FALSE)
+      if ((length(descending) > 1) && length(descending) != length(by)) {
+        stop(paste0("'descending' must be the same length as 'by' if more than one entry is supplied. Got 'by' with length ", length(by), " and 'descending' with length ", length(descending), "."))
+      }
+      if ((length(abs_sort) > 1) && length(abs_sort) != length(by)) {
+        stop(paste0("'abs_sort' must be the same length as 'by' if more than one entry is supplied. Got 'by' with length ", length(by), " and 'abs_sort' with length ", length(abs_sort), "."))
+      }
+      return(TableHandle$new(self$.internal_rcpp_object$sort(by, descending, abs_sort)))
+    },
+    
+    merge = function(...) {
+      table_list <- unlist(c(...))
+      if (length(table_list) == 0) {
+        return(TableHandle$new(self$.internal_rcpp_object$merge(NULL)))
+      }
+      verify_type("table_list", table_list, "TableHandle", "Deephaven TableHandle", FALSE)
+      unwrapped_table_list <- lapply(table_list, strip_r6_wrapping)
+      return(TableHandle$new(self$.internal_rcpp_object$merge(unwrapped_table_list)))
+    }
+  )
```

Comment:
I can go either way about having `merge` attached to the table handle.  We must have the function `merge` not associated with a table handle.  If this is kept in, confirm with works with null inputs.